packages/form-js-viewer/src/core/Validator.js

```diff
@@ -69,11 +69,17 @@ export default class Validator {
       ];
     }
 
-    if (validate.required && (isNil(value) || value === '')) {
-      errors = [
-        ...errors,
-        'Field is required.'
-      ];
+    if (validate.required) {
```

Comment:
The required radio does not work properly.

![image](https://user-images.githubusercontent.com/9433996/229731771-39341d05-6a1d-431b-b4c5-716fcbf0f75d.png)

1. Add a radio
2. Specify as required
3. No error is shown although no value is given

I think this is similar to other fields, [required is only triggering once the value got erased](https://github.com/bpmn-io/form-js/issues/323). That's not possible for the radio though 😅 So it would only show up on submit, so I think we are fine in that place.

Reply:
Maybe it makes sense to add tests to the `Validator.spec.js#required` to verify the new cases (checkbox, checklist, radio, taglist, datetime)?

Reply:
This is expected behavior, we shouldn't trigger the validation straight away as that would lead to prompting a user with errors before they have done anything.

We should validate on losing focus however, which we currently do not do, but that's an old thing that I'll tackle soon.

The proper way to test here is by supplying a submit button.